From dc33d4895caa2f54a101a7f1dd3cee8ac707a617 Mon Sep 17 00:00:00 2001
From: Francois Gaffie <francois.gaffie@intel.com>
Date: Fri, 23 Nov 2012 15:59:22 +0100
Subject: Add Stream Flags key to setParameters

BZ: 70126
[CLEAN]

Audio Route Manager needs to know the type of stream to take the
appropriate routing decision (eg direct Output, MP3 Offload).
It must be define as key and filled by the policy.

Category: device enablement
Domain: AudioComms-Common
Origin: internal
Upstream-Candidate: yes
Change-Id: I73035f27445109d758d1cd6b213fa93108be19c0
Orig-Change-Id: I32b477180173cf779576ca5c2a07db732d2c9979
Signed-off-by: Francois Gaffie <francois.gaffie@intel.com>
---
 include/media/AudioParameter.h    |    2 ++
 media/libmedia/AudioParameter.cpp |    1 +
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/include/media/AudioParameter.h b/include/media/AudioParameter.h
index 891bc4b..39e7bc1 100644
--- a/include/media/AudioParameter.h
+++ b/include/media/AudioParameter.h
@@ -41,6 +41,7 @@ public:
     //  keyInputSource: to change audio input source, value is an int in audio_source_t
     //     (defined in media/mediarecorder.h)
     //  keyScreenState: either "on" or "off"
+    //  keyStreamFlags: to give additional information to Audio HAL e.g. direct stream, offload
     static const char * const keyRouting;
     static const char * const keySamplingRate;
     static const char * const keyFormat;
@@ -48,6 +49,7 @@ public:
     static const char * const keyFrameCount;
     static const char * const keyInputSource;
     static const char * const keyScreenState;
+    static const char * const keyStreamFlags;
 #ifdef QCOM_HARDWARE
     static const char * const keyHandleFm;
     static const char * const keyVoipCheck;
diff --git a/media/libmedia/AudioParameter.cpp b/media/libmedia/AudioParameter.cpp
index e3fea77..6fd497a 100644
--- a/media/libmedia/AudioParameter.cpp
+++ b/media/libmedia/AudioParameter.cpp
@@ -32,6 +32,7 @@ const char * const AudioParameter::keyChannels = AUDIO_PARAMETER_STREAM_CHANNELS
 const char * const AudioParameter::keyFrameCount = AUDIO_PARAMETER_STREAM_FRAME_COUNT;
 const char * const AudioParameter::keyInputSource = AUDIO_PARAMETER_STREAM_INPUT_SOURCE;
 const char * const AudioParameter::keyScreenState = AUDIO_PARAMETER_KEY_SCREEN_STATE;
+const char * const AudioParameter::keyStreamFlags = AUDIO_PARAMETER_KEY_STREAM_FLAGS;
 #ifdef QCOM_HARDWARE
 const char * const AudioParameter::keyHandleFm = AUDIO_PARAMETER_KEY_HANDLE_FM;
 const char * const AudioParameter::keyVoipCheck = AUDIO_PARAMETER_KEY_VOIP_CHECK;
-- 
1.7.4.1

